conans/model/options.py

```diff
@@ -160,25 +163,34 @@ def __init__(self, values=None):
         if not values:
             return
 
+        if isinstance(values, six.string_types):
+            raise ConanException("String type not expected in OptionValues constructor, "
+                                 "did you mean to call OptionValues.loads(<string>)")
+
         # convert tuple "Pkg:option=value", "..." to list of tuples(name, value)
         if isinstance(values, tuple):
-            new_values = []
-            for v in values:
-                option, value = v.split("=", 1)
-                option, value = option.strip(), value.strip()
-                if not value:
-                    raise ConanException("Please define a default value for '%s' in"
-                                         "'default_options'" % option)
-                new_values.append((option, value))
-            values = new_values
+            values = [item.split("=", 1) for item in values]
+
+        # Check that we have a list of tuples
+        offenders = list(filter(lambda u: not isinstance(u, (tuple, list)) or len(u) <= 1, values))
+        if offenders:
+            offender_str = lambda u: str(u) if not isinstance(u, (tuple, list)) else u[0]
+            raise ConanException("Please define a default value for '%s' in "
+                                 "'default_options'" % "', '".join(map(offender_str, offenders)))
+
+        # Raise on repeated option
+        counter = Counter([k for k, v in values])
+        if len(counter.values()) != len(values):
+            raise ConanException("There is a repeated element in 'default_options'.")
 
         # handle list of tuples (name, value)
         for (k, v) in values:
+            k = k.strip()
+            v = v.strip() if isinstance(v, six.string_types) else v
             tokens = k.split(":")
             if len(tokens) == 2:
                 package, option = tokens
-                package_values = self._reqs_options.setdefault(package.strip(),
-                                                               PackageOptionValues())
+                package_values = self._reqs_options.setdefault(package, PackageOptionValues())
```

Comment:
beware, ``package`` can contains spaces after the split

Reply:
I need to add a test for this